frontend/src/metabase/setup/selectors.ts

```diff
@@ -120,25 +130,32 @@ export const getSteps = createSelector(
       !hasAddedPaidPlanInPreviousStep;
     const shouldShowDataUsageStep = !isHosted;
 
-    const steps: { key: SetupStep; isActiveStep: boolean }[] = [
-      { key: "welcome" as const },
-      { key: "language" as const },
-      { key: "user_info" as const },
-      { key: "usage_question" as const },
-      shouldShowDBConnectionStep && {
-        key: "db_connection" as const,
-      },
-      shouldShowLicenseStep && { key: "license_token" as const },
-      shouldShowDataUsageStep ? { key: "data_usage" as const } : null,
-      { key: "completed" as const },
-    ]
-      .filter(isNotFalsy)
-      .map(({ key }) => ({
-        key,
-        isActiveStep: activeStep === key,
-      }));
-
-    return steps;
+    const maybeAddStep = (step: SetupStep, condition: boolean): SetupStep[] =>
+      condition ? [step] : [];
+
+    const baseSteps: SetupStep[] = [
+      "welcome",
+      "language",
+      "user_info",
+      "usage_question",
+    ];
+    const completionStep: SetupStep[] = ["completed"];
+    const embeddingSteps: SetupStep[] = ["user_info", ...completionStep];
+
+    const steps = isEmbeddingUseCase
+      ? embeddingSteps
+      : [
+          ...baseSteps,
+          ...maybeAddStep("db_connection", shouldShowDBConnectionStep),
+          ...maybeAddStep("license_token", shouldShowLicenseStep),
+          ...maybeAddStep("data_usage", shouldShowDataUsageStep),
+          ...completionStep,
+        ];
```

Comment:
nice! like this setup. One thing I would change: I don't see a need to add an abstraction on top of `baseSteps` and `completionStep` - there can maybe just be `normalSteps` and `embeddingSteps` - if we end up with more branches, let someone else come up with a sensible abstraction at that point.

Reply:
https://github.com/metabase/metabase/pull/57749/commits/77a037016cf9bd06a1a1511ea6e635b8ba6ea933